docs/development/pigs/pig-001.rst

```diff
@@ -0,0 +1,120 @@
+.. include:: ../../references.txt
+
+.. _pig-001:
+
+**********************************
+PIG 1 - PIG purpose and guidelines
+**********************************
+
+* Author: Christoph Deil
+* Created: December 20, 2017
+* Status: draft
+
+What is a PIG?
+==============
+
+Proposals for improvement of Gammapy (PIGs) are short documents proposing a major
+addition or change to Gammapy.
+
+PIGs are like `APEs`_, `PEPs`_, `NEPs`_ and `JEPs`_, just for Gammapy.
+Using such enhancement proposals is common for large and long-term
+open-source Python projects.
+
+The primary goal of PIGs is to have an open
+and structured way of working on Gammapy, forcing the person making the
+change to think it through and motivate the proposal before taking action,
+and for others to have a chance to review and comment on the proposal. The PIGs will also
+serve as a record of major design decisions taken in Gammapy, which can
+be useful in the future when things are re-discussed or new proposals to
+do things differently arrive.
+
+We expect that we will not use PIGs very often, but we think they can be useful
+e.g. in the following cases:
+
+* Outline design for something that requires significant work,
+  e.g. on topics like "Modeling in Gammapy" or "High-level user interface in Gammapy".
+  These PIGs can be rather long, i.e. more than one page, explaining the
+  design in detail and even explaining which alternatives were
+  considered and why the proposed solution was preferred.
+* Have a conscious decision and make sure all interested parties
+  are aware for things that might be controversial and have long-term
+  effects for Gammapy, e.g. when to drop Python 2 support or defining
+  a release cycle for Gammapy. These PIGs can usually be very short, a page or less.
+
+Writing a PIG
+=============
+
+Anyone is welcome to write a PIG!
+
+PIGs are written as RST files in the ``docs/development/pigs`` folder in the
+main Gammapy repository, and submitted as pull requests.
+
+Most discussions concerning Gammapy will happen by talking to each other directly
+(calls or face-to-face), or online on the mailing list or Github.
+If you're not sure if you should write a PIG, please don't! Instead bring the
+topic up in discussions first with other Gammapy developers, or on the mailing list,
+to get some initial feedback. This will let you figure out if writing a PIG
+will be helpful or not to realise your proposal.
+
+For now, we don't require a formal structure, apart from adding the author,
```

Comment:
Even if we don't **require** a formal structure I think it might actually help the authors to provide a minimal template, even if it only contains the top section.

Reply:
I think the structure is different for different PIGs. I can add a sentence pointing to nice examples of 2-3 PEPs or APEs that people can look at for guideline what a high-quality and well-structured proposal can look like.

Reply:
> I can add a sentence pointing to nice examples of 2-3 PEPs or APEs that people can look at for guideline what a high-quality and well-structured proposal can look like.

:+1:

Reply:
I've added a section here describing structure a bit and pointing to example APEs and PEPs.